src/session/sidebar/row.rs

```diff
@@ -359,9 +370,78 @@ impl Row {
+                    &[
+                        actions_expression.upcast_ref(),
+                        draft_message_expression.upcast_ref(),
+                        last_message_expression.upcast_ref(),
+                    ],
+                    closure!(|_: Chat,
+                              actions: ChatActionList,
+                              draft_message: Option<BoxedDraftMessage>,
+                              message: Option<Message>| {
+                        actions
+                            .last()
+                            .map(|_| String::new())
+                            .or_else(|| draft_message.map(|_| String::new()))
+                            .or_else(|| {
+                                message
+                                    .and_then(sender_label)
+                                    .map(|sender_label| format!("{sender_label}:"))
+                            })
+                    }),
+                )
+                .bind(&*imp.last_message_sender_label, "label", Some(chat));
+                bindings.push(sender_binding);
+
+                let thumbnail_binding = gtk::ClosureExpression::new::<Option<gdk::Texture>, _, _>(
+                    &[
+                        actions_expression.upcast_ref(),
+                        draft_message_expression.upcast_ref(),
+                        last_message_expression.upcast_ref(),
+                    ],
+                    closure!(|_: Chat,
+                              actions: ChatActionList,
+                              draft_message: Option<BoxedDraftMessage>,
+                              message: Option<Message>| {
+                        if actions.n_items() > 0 || draft_message.is_some() {
+                            None
+                        } else {
+                            message.and_then(|message| match message.content().0 {
+                                MessageContent::MessagePhoto(data) => {
+                                    data.photo.minithumbnail.map(|thumb| {
+                                        let pixbuf = gdk::pixbuf_get_from_texture(
+                                            &gdk::Texture::from_bytes(&glib::Bytes::from_owned(
+                                                glib::base64_decode(&thumb.data),
+                                            ))
+                                            .unwrap(),
+                                        )
+                                        .unwrap();
+
+                                        let target_size = thumb.width.max(thumb.height);
+
+                                        let pixbuf_scaled = pixbuf
+                                            .scale_simple(
+                                                target_size,
+                                                target_size,
+                                                gdk_pixbuf::InterpType::Bilinear,
+                                            )
```

Comment:
I'd really avoid doing that expensive operation honestly. Also I don't really want to use `GdkPixbuf` as that API is meant to go away in the future. What we need here is to something similar to css's [object-fit: cover](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) which isn't available natively in GTK widgets (I know that there was some discussion to add that, but no action was taken). So... we have to implement our own widget for now.

Fortunately, the `MediaPicture` widget is pretty close to what we need. You just need to remove the `aspect-ratio` property, since we need a `1:1` widget, then you should use the paintable intrinsic aspect ratio inside the snapshot virtual function. Also, we don't really need the measure and request_mode thing, since we just need to set minimum sizes via `width-request` and `height-request`.

Reply:
GitHub is drunk and it decided to post a double comment.